lightning/src/ln/channel.rs

```diff
@@ -5254,6 +5254,13 @@ impl<Signer: Sign> Writeable for Channel<Signer> {
 			htlc.write(writer)?;
 		}
 
+		// If the channel type is something other than only-static-remote-key, then we need to have
+		// older clients fail to deserialize this channel at all. If the type is
+		// only-static-remote-key, we simply consider it "default" and don't write the channel type
+		// out at all.
+		let serialized_chan_type = if self.channel_type != ChannelTypeFeatures::only_static_remote_key() {
```

Comment:
nit: To me "serialized" means something already serialized to bytes. Maybe just call this `channel_type`?